### PR TITLE
chore: Bump Buf to v1.29.0 in GHA

### PR DESCRIPTION
**Pull request by codingllama**
Missed by #37429.

### CODE REVIEW

**Review by tigrato — APPROVED** [review]

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -70,7 +70,7 @@ jobs:
       - uses: bufbuild/buf-setup-action@382440cdb8ec7bc25a68d7b4711163d95f7cc3aa # v1.28.1
         with:
           github_token: ${{ github.token }}
-          version: v1.28.0
+          version: v1.29.0
```
[the remark this answers is not in the document]
Comment (codingllama) [reply .github/workflows/lint.yaml:73]: Good question. It looks like they just bumped version numbers, but I don't see any harm in touching it up here too - https://github.com/bufbuild/buf-setup-action/compare/v1.28.1...v1.29.0.

I would expect the bots would automatically try to bump this, so I'll take a look at that too.

[the remark this answers is not in the document]
Comment (codingllama) [reply .github/workflows/lint.yaml:73]: Done.

[the remark this answers is not in the document]
Comment (codingllama) [reply .github/workflows/lint.yaml:73]: Mystery solved: https://github.com/gravitational/teleport/pull/37442/files.

**Review by wadells — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

**Comment by codingllama** [comment]
Thanks for the quick reviews!

**Pull request merged**